src/contracts/libraries/GPv2SafeERC20.sol

```diff
@@ -15,10 +13,22 @@ library GPv2SafeERC20 {
         address to,
         uint256 value
     ) internal {
-        _callOptionalReturn(
-            token,
-            abi.encodeWithSelector(token.transfer.selector, to, value)
-        );
+        bytes4 selector_ = token.transfer.selector;
+
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            let freeMemoryPointer := mload(0x40)
+            mstore(freeMemoryPointer, selector_)
+            mstore(add(freeMemoryPointer, 4), to)
+            mstore(add(freeMemoryPointer, 36), value)
+
+            if iszero(call(gas(), token, 0, freeMemoryPointer, 68, 0, 0)) {
+                returndatacopy(0, 0, returndatasize())
+                revert(0, returndatasize())
+            }
```

Comment:
> Does call create a completely fresh memory stack and copies the input memory to it?

AFAIK, it creates a fresh "execution context" with its own memory. Otherwise calling other smart contracts could write on your memory and break things (a malicious interaction would write on the `TradeExecution.Data.exeutedBuyAmount = 0` and have the settlement contract keep all the fees! Muahahahaha).

> Can you add a test with very large return data (at least overwriting the free memory pointer)?

Yes! That's a very good point, I don't cover that branch.

Reply:
Added a test!